x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResponseListener.java

```diff
@@ -154,6 +154,10 @@ private RestResponse buildResponse(EsqlQueryResponse esqlResponse) throws IOExce
                 );
             }
             long tookNanos = stopWatch.stop().getNanos();
+
+            // TODO: question for reviewers - should we instead grab took time from the EsqlQueryResponse now and remove this StopWatch?
+            // TODO: that would solve the issue https://github.com/elastic/elasticsearch/issues/110935
+            // TimeValue tookMillis = esqlResponse.getExecutionInfo().getOverallTook();
             restResponse.addHeader(HEADER_NAME_TOOK_NANOS, Long.toString(tookNanos));
```

Comment:
Yes. I think we should.

Reply:
Thanks Nik. I think this means then that I will need to change the code in this PR to track `took` time in nanos, rather than millis? Or do we want to remove/deprecate this header and put in one just called `took`? (I have a similar question out to Product, but would like to hear your views.)

Reply:
I've updated the internal to track took time by nanos now. However, I left the XContent output in millis, as that is how _search does it. Question is out to @naj-h whether to leave this as millis to change to `took_nanos` in the response object. (The `Took-nanos` header is still present and done in nanos.)